### ISSUE TITLE
[boost_stacktrace] No handling of different backends

### ISSUE THREAD

**Issue by theodelrieu** [bincrafters/community#128]
### Description of Problem, Request, or Question

Boost.Stacktrace can be header-only or compiled separately.

When using the header-only version, one needs to define a macro to choose which backend they want to use.

For the compiled version, one has to link against a different library each time.

Note that the library chooses a default macro, but only in the header-only version.
It also means that using `tools.collect_libs` in the case of Boost.Stacktrace is not the good solution.

I believe adding an option with multiple string values (one for each backend, plus providing a default depending on the host platform) would be great.

More information can be found at the Boost.Stacktrace [Configuration and Build page](http://www.boost.org/doc/libs/1_65_1/doc/html/stacktrace/configuration_and_build.html).

### Package Details
* Package Name/Version: Boost.Stacktrace/1.65.[0,1], boost_stacktrace/1.66.0

**Comment by grafikrobot** [comment]
I would prefer if we minimize the options we provide and prefer using the general conan profile settings. As it’s more maintenance and can get us in trouble in the long run as those options change because the upstream changed their implementation.

Since, believe it or not, I'm not a conan user I also don't understand what the issue is with using `collect_libs`. Does it prevent a use case?

**Comment by solvingj** [comment]
I can see this library comes with a formidable amount of complexity in the handling of compilation.  It does seem that it would/will require extensive options to model properly in Conan.  Meanwhile, it's a header-only library by default, and are currently not treating it as header-only.  

I suggest first adding the option (default=true) to build header-only, and change CI to treat this as header-only.  

Once that is complete, we can begin work on modeling all the platform-specifics into Conan with options and environment detection.  By "we", I mean this will require an actual user of the library and Conan who truly wants to use this functionality to contribute such a PR.  It will most certainly involve many hours of research and trial/error.

**Comment by theodelrieu** [comment]
@grafikrobot 

>I also don't understand what the issue is with using collect_libs. Does it prevent a use case?

On Linux, the link order is very important. One must link against `-lfoo -lbar` if `libbar` is a dependency of `libfoo`. `tools.collect_libs` does not fiddle with link order at all.

For this specific library, you have to link against a single static library, and `collect_libs` will make you link against all of them.

@solvingj 

Indeed, the library looks very complex. Supporting every backend looks hard, especially `libbacktrace` which has not been updated for a year.

I also think that beginning with header-only support is the best solution right now.

**Comment by ArekPiekarz** [comment]
@theodelrieu libbacktrace has received updates recently. I just tested it with Boost.Stacktrace 1.66 from Msys2 and it now works correctly (see my last comment in https://github.com/boostorg/stacktrace/issues/41).

**Comment by datalogics-kam** [comment]
This is interesting to me as well, since by default on Windows, `boost_stacktrace` results in duplicate symbols. I was able to get it to work by defining, in CMake:

```
target_compile_definitions(util PUBLIC BOOST_STACKTRACE_LINK BOOST_STACKTRACE_USE_WINDBG)
```

It would be nice to have options that configure boost_stacktrace.

**Comment by solvingj** [comment]
Closing as too difficult to support.  Enterprise/professional users can fork and add support for this as needed.
